src/common/cockpittest.c

```diff
@@ -344,6 +344,9 @@ _cockpit_assert_strmatch_msg (const char *domain,
   gchar *msg;
   int len;
 
+  if (pattern == NULL)
+    pattern = "";
```

Comment:
I feel this is a bit too magic for unit tests -- I'd rather the caller would explicitly call it with a `my_pattern || ""` to clearly mark the places where a NULL value is okay (like unset fields in the TestCase expected stdout/stderr)

Reply:
The reason I do this is it avoids me having to explicitly set a pattern in the TestCase struct for all of the cases where the empty string is expected (since NULL is the default there).

Maybe you're right though.  It won't kill me to be explicit there.

Reply:
No, I meant in `test_check()` -- they could call assert_strmatch with `tc->check_stdout || ""`